lmms_eval/protocol.py

```
@@ -114,3 +127,122 @@ def encode_image(self, image: Union[Image.Image, str]):
+    def get_video_length(self, video_path: str, return_nframes: bool = False) -> float:
+        try:
+            vr = VideoReader(video_path)
+            if return_nframes:
+                return len(vr) / vr.get_avg_fps(), len(vr)
+            else:
+                return len(vr) / vr.get_avg_fps()
+        except Exception as e:
+            print(f"Error getting video length for {video_path}: {e}")
+            return -1
+
+    def process_video(self, video_path: str, max_pixels: int, fps: int, start_time: float = 0, end_time: float = None, nframes: int = None):
```

Ruff (0.12.2)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)